Mathlib/Data/Fintype/Basic.lean

```diff
@@ -141,12 +141,71 @@ theorem codisjoint_left : Codisjoint s t ↔ ∀ ⦃a⦄, a ∉ s → a ∈ t :=
 theorem codisjoint_right : Codisjoint s t ↔ ∀ ⦃a⦄, a ∉ t → a ∈ s :=
   Codisjoint_comm.trans codisjoint_left
 
-section BooleanAlgebra
+instance booleanAlgebra [DecidableEq α] : BooleanAlgebra (Finset α) :=
+  GeneralizedBooleanAlgebra.toBooleanAlgebra
```

Comment:
What's the reason for moving this up and outside of the `BooleanAlgebra` section?

Reply:
I need `sᶜ` to make sense to write the elaborator and I want to be able to use the elaborator in the rest of the section, so I had to cut the section in half with the elaborator in the middle,

Reply:
Ah yes, that makes sense. Didn't realise that `sᶜ` requires the `BooleanAlgebra` instance.